### PR TITLE
(Old) ETA Qatar changes 

### PR DESCRIPTION
**Pull request by cynthia-anya**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: app/flows/check_uk_visa_flow/outcomes/outcome_marriage_electronic_visa_waiver.erb [hunk 1]
```diff
@@ -11,5 +11,9 @@
   - settled in the UK
   - someone with humanitarian protection, for example if they have refugee status
 
-  ^If you want to convert a civil partnership into a marriage, you'll need to apply for an [electronic visa waiver (EVW)](/get-electronic-visa-waiver) or a [Standard Visitor visa](/standard-visitor).
+  If you want to convert a civil partnership into a marriage, you'll need to apply for an [electronic visa waiver (EVW)](/get-electronic-visa-waiver) or a [Standard Visitor visa](/standard-visitor).
+
+  <%if calculator.passport_country_requires_electronic_travel_authorisation?  %>
```
Comment (gclssvglx) [inline app/flows/check_uk_visa_flow/outcomes/outcome_marriage_electronic_visa_waiver.erb:16]: Surprised rubocop hasn't flagged this, but the house-style is `<% if ... %>` - so one space after `%` and one before.

File: app/flows/check_uk_visa_flow/outcomes/outcome_no_visa_needed.erb [hunk 2]
```diff
@@ -3,6 +3,10 @@
 <% end %>
 
 <% govspeak_for :body do %>
+  <%if calculator.passport_country_requires_electronic_travel_authorisation?  %>
+    <%= render partial: 'electronic_travel_authorisation_advice', locals: {calculator: calculator} %>
+  <% end %>
+
   <% if (calculator.travelling_to_elsewhere? || calculator.travelling_to_ireland?) &&
```
Comment (gclssvglx) [inline app/flows/check_uk_visa_flow/outcomes/outcome_no_visa_needed.erb:10]: I wonder if formatting this a little and giving it extra brackets would help it's readability... 🤔 

```ruby
<% if (
    (calculator.travelling_to_elsewhere? || calculator.travelling_to_ireland?) && 
    (calculator.passport_country_in_non_visa_national_list? || 
        calculator.passport_country_in_eea? || 
        calculator.passport_country_in_british_overseas_territories_list?)
) %>
```

File: lib/smart_answer/calculators/uk_visa_calculator.rb [hunk 3]
```diff
@@ -78,6 +78,10 @@ def passport_country_in_electronic_visa_waiver_list?
       COUNTRY_GROUP_ELECTRONIC_VISA_WAIVER.include?(@passport_country)
     end
 
+    def passport_country_requires_electronic_travel_authorisation?
+      %w[qatar].include?(@passport_country)
+    end
```
Comment (gclssvglx) [inline lib/smart_answer/calculators/uk_visa_calculator.rb:83]: Nice 👍 - this will really help later when more countries are added 🥇

**Review by gclssvglx — COMMENTED** [review]
This is looking really good @cynthia-anya - just a few picky suggestions.

File: app/flows/check_uk_visa_flow/outcomes/outcome_marriage_electronic_visa_waiver.erb (hunk 1 above)
Comment (sivsubra-tw) [reply app/flows/check_uk_visa_flow/outcomes/outcome_marriage_electronic_visa_waiver.erb:16]: We will need [erb-lint](https://github.com/Shopify/erb-lint) for it which I can add in a separate PR. For now, will add the space manually 👍

File: app/flows/check_uk_visa_flow/outcomes/outcome_no_visa_needed.erb (hunk 2 above)
Comment (sivsubra-tw) [reply app/flows/check_uk_visa_flow/outcomes/outcome_no_visa_needed.erb:10]: IDE complained about having parentheses around a conditional. But I am incorporating the breaking of lines as you have suggested 👍

**Review by gclssvglx — APPROVED** [review]
Great stuff - looks good to go 👍

**Pull request closed without merging**